### ISSUE TITLE
Incoming SMS during Restore

### ISSUE THREAD

**Issue by imnes** [jberkel/sms-backup-plus#809]
If I'm doing a long running restore, say it takes a few hours.  While that's running and SMS Backup is set as the default messaging app, will new incoming texts get dropped / lost?

**Comment by Woi** [comment]
I don't know the answer. But you could just enable flight mode and re-enable wifi till the restore has finished.

**Comment by nbali** [comment]
I have a feeling that it's lost. At least for me the default app wasn't restored, and I went days without any SMS. It might be coincidence, but I suspect it's not.

ps.: successfully duplicated this one before finding it with https://github.com/jberkel/sms-backup-plus/issues/851

**Comment by jberkel** [comment]
yes, there's a chance that it can get lost. this is unfortunately a result of Android's permission design, which requires the app to become the default SMS app in order to write to the database. everytime you start SMS Backup it will check if it's still the default app and ask you to switch back.

I'm sure the process could be improved, but as I said there are substantial design flaws in the OS itself.

https://github.com/jberkel/sms-backup-plus/#faq-restore-default-app

**Comment by imnes** [comment]
Could functionality be added to allow sms-backup-plus to continue to
receive and store texts as unread while it's active?  Then when another app
is restored as default client they'd show up in that app?

**Comment by jberkel** [comment]
@imnes I need to check – it should be possible. but there's always a risk of something going wrong (a bug in the receiving code etc), that's why enabling flightmode during long restore operations seems to be the safest choice right now

**Comment by nbali** [comment]
Even if that's not possible, a warning pop-up about this would be useful if the phone isn't in flightmode. (Obviously a prompt to turn it on would be best, but I have no idea if it's doable.)

**Comment by imnes** [comment]
I would think there's an android SMS library that can be utilized but
haven't done much research yet.  If I get some free time soon I'll look
into some options and what it would take to integrate!

**Comment by jberkel** [comment]
I'll add a more prominent warning. There's already a popup, but it doesn't mention the risk of data loss.

**Comment by jberkel** [comment]
@imnes  @nbali will be fixed in next beta (-14)

**Comment by jberkel** [comment]
@nbali @imnes I'd be cool if you could verify that this actually works. If have only tested on an emulator.

**Comment by imnes** [comment]
Did you add sms receipt or that notice?

**Comment by jberkel** [comment]
@imnes it saves the incoming SMS to the storage. the notice was updated as well.

**Comment by imnes** [comment]
Awesome! Will give it a try and report back!

**Comment by jcrben** [comment]
@imnes can you try? would be good to get confirmation before 1.5.11

**Comment by imnes** [comment]
Is it in the current release in app store or do I need to sign up for a
beta or something?

**Comment by jcrben** [comment]
https://github.com/jberkel/sms-backup-plus#beta

**Comment by imnes** [comment]
I installed the latest beta last night and attempted a backup overnight.
When I woke up it had shown an I/O error, not sure if I'll have a chance to
test restore functionality before next week.
